test/ut/wrappers/unique.cpp

```diff
@@ -17,8 +17,13 @@ struct interface {
   virtual ~interface() noexcept = default;
   virtual void dummy() = 0;
 };
-struct implementation : public interface {
-  void dummy() override{};
+struct implementation : interface {
+  implementation() = default;
+  explicit implementation(int f) : f(f) {}
+  implementation(implementation&& other) noexcept = default;
+  implementation(implementation&) = delete;
+  void dummy() override {}
+  int f = 0;
```

Comment:
This set of constructors is quite ugly but I didn't come up with anything better so far